kolibri/plugins/learn/assets/src/composables/useDownloadRequests.js

```diff
@@ -27,107 +31,75 @@ const downloadRequestMap = reactive({
   downloads: {},
   totalPageNumber: 0,
   totalDownloads: 0,
+  totalStorage: 0,
 });
 
 export default function useDownloadRequests(store) {
   store = store || getCurrentInstance().proxy.$store;
   function fetchUserDownloadRequests(params) {
     const { page, pageSize } = params;
     const loading = ref(true);
-    const dummyDownloadRequests = [
-      {
-        id: '2ea9bda8703241be89b5b9fd87f88815',
-        user_id: store.getters.currentUserId,
-        reason: 'USER_INITIATED',
-        facility_id: store.getters.userFacilityId,
-        status: 'QUEUED',
-        date_added: new Date(),
-        resource_metadata: {
-          title: 'Intro to addition',
-          file_size: 1113580,
-          learning_activities: ['UD5UGM0z'],
-        },
-        node_id: '2ea9bda8703241be89b5b9fd87f88815',
-      },
-      {
-        id: '9e53d545aaf44c3787a29a34b189c56a',
-        user_id: store.getters.currentUserId,
-        reason: 'USER_INITIATED',
-        facility_id: store.getters.userFacilityId,
-        status: 'QUEUED',
-        date_added: new Date(),
-        resource_metadata: {
-          title: 'PDF 1 page',
-          file_size: 3113580,
-          learning_activities: ['wA01urpi'],
-        },
-        node_id: '9e53d545aaf44c3787a29a34b189c56a',
-      },
-    ];
-    setTimeout(() => {
+    let storage = 0;
+    return ContentDownloadRequestResource.list(params).then(downloadRequests => {
       set(downloadRequestMap, 'downloads', {});
       for (let i = 0; i < pageSize; i++) {
         const index = (page - 1) * pageSize + i;
-        if (index >= dummyDownloadRequests.length) {
+        if (index >= downloadRequests.length) {
           break;
         }
-        set(
-          downloadRequestMap.downloads,
-          dummyDownloadRequests[index].node_id,
-          dummyDownloadRequests[index]
-        );
+        storage += downloadRequests[index].metadata
+          ? downloadRequests[index].metadata.file_size
+          : storage;
+        set(downloadRequestMap.downloads, downloadRequests[index].id, downloadRequests[index]);
       }
-      set(
-        downloadRequestMap,
-        'totalPageNumber',
-        Math.ceil(dummyDownloadRequests.length / pageSize)
-      );
-      set(downloadRequestMap, 'totalDownloads', dummyDownloadRequests.length);
+      set(downloadRequestMap, 'totalPageNumber', Math.ceil(downloadRequests.length / pageSize));
+      set(downloadRequestMap, 'totalDownloads', downloadRequests.length);
+      set(downloadRequestMap, 'totalStorage', storage);
       set(loading, false);
     }, 500);
-    return loading;
   }
 
   function fetchDownloadsStorageInfo() {
     const loading = ref(true);
     const storageInfo = ref(null);
-    const dummyStorageInfo = {
-      freeDiskSize: 13340000000,
-      myDownloadsSize: 23200000,
-    };
+    let freespace = 0;
+    client({
+      url: `${urls['kolibri:core:freespace']()}`,
```

Comment:
This API endpoint is only accessible to those with `canManageContent` permissions currently:

https://github.com/learningequality/kolibri/blob/develop/kolibri/core/device/api.py#L110

You should update this to an ORed permissions class with: `CanManageContent | UsersCanDownloadResources` with the latter gated by the relevant device permission.

(this can be a follow up issue if you so choose)